persistence/db.go

```diff
@@ -65,31 +68,51 @@ func (pg *PostgresContext) ResetContext() error {
 	return nil
 }
 
-// TECHDEBT: Implement proper connection pooling
-func connectToDatabase(postgresUrl string, schema string) (*pgx.Conn, error) {
+func connectToDatabase(cfg modules.PersistenceConfig) (*pgx.Conn, error) {
 	ctx := context.TODO()
 
-	conn, err := pgx.Connect(context.Background(), postgresUrl)
+	config, err := pgxpool.ParseConfig(cfg.GetPostgresUrl())
+	if err != nil {
+		return nil, fmt.Errorf("unable to create database config: %v", err)
+	}
+	MaxConnLifetime, err := time.ParseDuration(cfg.GetMaxConnLifetime())
+	if err == nil {
+		config.MaxConnLifetime = MaxConnLifetime
+	}
+	MaxConnIdleTime, err := time.ParseDuration(cfg.GetMaxConnIdleTime())
```

Comment:
Let's return an error if parsing fails/

I understand the logic of "a quite success" of using the defaults, but it could lead to a bug where someone incorrectly configures a time parameter and spends hours debugging why it wasn't applied.